src/main/java/games/strategy/engine/data/TripleaHandler.java

```diff
@@ -0,0 +1,475 @@
+package games.strategy.engine.data;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Deque;
+import java.util.LinkedList;
+
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.SAXParseException;
+import org.xml.sax.helpers.AttributesImpl;
+import org.xml.sax.helpers.DefaultHandler;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+
+/**
+ * Abstract common implementation of a DefaultHandler.
+ */
+public abstract class TripleaHandler extends DefaultHandler {
+
+  public static final String XML_NODE_NAME_ALLIANCE = "alliance";
```

Comment:
Using an enum instead of String I think would help simplify quite a bit. I'm not sure if you'll be able to capture the parent-child relationship more cleanly, but there will be at least a strong type for node names, we won't need the redundant `XML_NODE_NAME` prefix anymore, and we'll have a strong value type instead of string and can still capture the `label -> string` relationship.